### PR TITLE
Exhaustiveness checks

### PR DESCRIPTION
**Pull request by jez**
### Test plan

See included automated tests.

### CODE REVIEW

**Comment by jez** [comment]
Screen Shot 2019-07-18 at 8 36 31 PM

File: cfg/Instructions.h [hunk 1]
```diff
@@ -187,6 +187,18 @@ class Cast final : public Instruction {
...
+public:
+    VariableUseSite what;
+
+    TImpossible(core::LocalVariable what) : what(what) {
+        categoryCounterInc("cfg", "timpossible");
+    }
+
+    virtual std::string toString(core::Context ctx);
+};
+// CheckSize(Cast, 40, 8);
```
Comment (DarkDimius) [inline cfg/Instructions.h]: either delete or uncomment

File: cfg/proto/proto.cc [hunk 2]
```diff
@@ -22,58 +22,149 @@ com::stripe::rubytyper::LocalVariable Proto::toProto(const core::GlobalState &gs
     return proto;
 }
 
-com::stripe::rubytyper::Instruction Proto::toProto(const core::GlobalState &gs, const Instruction *instr) {
-    com::stripe::rubytyper::Instruction proto;
+vector<com::stripe::rubytyper::Binding> Proto::toProto(const core::GlobalState &gs, const Binding &bnd) {
```
Comment (DarkDimius) [inline cfg/proto/proto.cc]: WDYT about `InlinedVector<com::stripe::rubytyper::Binding, 1>`?

File: website/docs/exhaustiveness.md [hunk 3]
```diff
@@ -3,15 +3,12 @@ id: exhaustiveness
 title: Exhaustiveness Checking
 ---
 
-> **Note**: Sorbet does not currently have first-class support for
-> exhaustiveness checks. This feature is planned, but has not yet been
-> implemented. Here we document a current workaround.
+**Exhaustiveness checking** is a feature of a language where the type checker
```
Comment (DarkDimius) [inline website/docs/exhaustiveness.md]: WDYT about separating docs update and landing them when we verified the feature on a few codebases?

**Pull request merged**